ludwig/data/split.py

```diff
@@ -69,21 +103,23 @@ def __init__(self, probabilities: List[float] = DEFAULT_PROBABILITIES, **kwargs)
     def split(
         self, df: DataFrame, backend: Backend, random_seed: float = default_random_seed
     ) -> Tuple[DataFrame, DataFrame, DataFrame]:
+        probabilities = _make_fractions_ensure_minimum_rows(self.probabilities, len(df))
```

Comment:
`len(df)` is pretty expensive in Dask (requires a full pass over the data). Maybe we can only perform this optimization for Pandas for the time being?

Reply:
This should be ok, as long as the distributed tests have enough rows.

Reply:
Actually, I'm going to take a stab at implementing this for the dask ddf case and see how it goes.  There are some distributed tests that could be fixed by guaranteeing sufficient split size.

Reply:
Implementing this in the distributed case is a lot more complicated...  so I decided to revert this PR back to local-only.  Will try to find a better solution for distributed and do that in a separate PR.